### PR TITLE
feat: hide remove attachment button if action can't be executed and check for team permissions

### PR DESCRIPTION
**Pull request by fmartingr**
#### Summary
- Check permissions and post edit time limit in the webapp as well.
- Check team permissions

### CODE REVIEW

**Review by jwilander — APPROVED** [review]

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:

File: webapp/src/index.tsx [hunk 1]
```diff
@@ -22,8 +25,28 @@ export default class Plugin {
...
+                // Check if post has attachments
+                if (!(typeof post.file_ids?.length !== 'undefined' && post.file_ids?.length > 0)) {
+                    return false;
+                }
+
+                // Check if the user has permissions to edit his own post or edit other's posts if not the author
+                const user = getCurrentUser(state);
+                let permission = 'edit_post';
+                if (post.user_id !== user.id) {
+                    permission = 'delete_others_posts';
```
Comment (saturninoabril) [inline webapp/src/index.tsx]: Should this be `edit_others_posts`?

nit: you may import the constants, so you could do:

```javascript
import {Permissions} from 'mattermost-redux/constants';

...
let permission = Permissions.EDIT_POST;
...
permission = Permissions.EDIT_OTHERS_POSTS;
```

File: webapp/src/index.tsx [hunk 2]
```diff
@@ -22,8 +25,28 @@ export default class Plugin {
...
+                }
+                return haveIChannelPermission(state, {
+                    channel: post.channel_id,
+                    permission,
+                });
```
Comment (saturninoabril) [inline webapp/src/index.tsx]: Works with channel permission but not those with team permission.

**Review by saturninoabril — CHANGES_REQUESTED** [review]
@fmartingr Tested but:
- not working if permission allows edit to others' post but not delete, and
- not working if permission is granted to team admin but not to channel admin.

File: webapp/src/index.tsx (hunk 2 above)
Comment (fmartingr) [reply webapp/src/index.tsx]: We are not checking team permissions on the backend either, so this shouldn't be working before either. let me make the checks on par.

File: webapp/src/index.tsx (hunk 1 above)
Comment (fmartingr) [reply webapp/src/index.tsx]: I was looking for that yesterday but couldn't find it. Thank you!

**Comment by fmartingr** [comment]
> @fmartingr Tested but:
> 
> * not working if permission allows edit to others' post but not delete, and

This is working for me:

https://github.com/user-attachments/assets/aca22593-b9fc-4454-8878-ce50fd07e451


> * not working if permission is granted to team admin but not to channel admin.

This was not checked on the backend either. Adding it now.

File: webapp/src/index.tsx (hunk 2 above)
Comment (saturninoabril) [reply webapp/src/index.tsx]: Thanks all good now!

**Review by saturninoabril — APPROVED** [review]
Thanks @fmartingr! Tested and all good to me now.

**Review by jwilander — APPROVED** [review]

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:

**Pull request merged**